modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/portlet/action/GetDataMVCResourceCommand.java

```
@@ -93,33 +96,30 @@ protected void doServeResource(
 		HttpServletRequest httpServletRequest = _portal.getHttpServletRequest(
 			resourceRequest);
 
-		LayoutDisplayPageObjectProvider<Object>
-			layoutDisplayPageObjectProvider =
-				(LayoutDisplayPageObjectProvider<Object>)
-					LayoutDisplayPageProviderUtil.
-						initLayoutDisplayPageObjectProvider(
-							httpServletRequest,
-							_layoutDisplayPageProviderTracker, _portal);
-
-		if (layoutDisplayPageObjectProvider == null) {
-			JSONPortletResponseUtil.writeJSON(
-				resourceRequest, resourceResponse,
-				JSONUtil.put(
-					"error",
-					_language.get(
-						httpServletRequest, "an-unexpected-error-occurred")));
-
-			return;
-		}
+		InfoItemClassPKReference infoItemClassPKReference =
+			_getInfoItemClassPKReference(httpServletRequest);
 
 		try {
+			Object analyticsReportsInfoItemObject = Optional.ofNullable(
+				_analyticsReportsInfoItemObjectProviderTracker.
+					getAnalyticsReportsInfoItemObjectProvider(
+						infoItemClassPKReference.getClassName())
+			).map(
+				analyticsReportsInfoItemObjectProvider ->
+					analyticsReportsInfoItemObjectProvider.
+						getAnalyticsReportsInfoItemObject(
+							infoItemClassPKReference)
+			).orElseThrow(
+				() -> new NoSuchModelException(
```

Do we want to throw an exception? Just asking because previously we did not

---

ohh, It's catched it down below, forget it